### PR TITLE
chore: add types to functions and correct docstrings

### PR DESCRIPTION
**Pull request by c0nsol3**
https://github.com/Solar-network/python-crypto/issues/44

I think I've covered all errors raised in issue. I've scanned over all other function definitions and checked for more mistakes and while I was at it, I've also added types to. Let me know in case I overdid it by adding type into this PR.

### CODE REVIEW

**Comment by sleepdefic1t** [comment]
:excellent:

Thank you, @c0nsol3 
We'll begin review on this soon.

File: solar_crypto/identity/private_key.py [hunk 1]
```diff
@@ -46,7 +48,7 @@ def from_passphrase(cls, passphrase):
         return cls(private_key)
 
     @classmethod
-    def from_hex(cls, private_key):
+    def from_hex(cls, private_key) -> PrivateKey:
```
Comment (sleepdefic1t) [inline solar_crypto/identity/private_key.py:51]: Shouldn't `private_key` also have a type here?

File: solar_crypto/transactions/builder/base.py
Comment (sleepdefic1t) [inline solar_crypto/transactions/builder/base.py:1]: The following lines have functions that appear to need types as well:

- https://github.com/Solar-network/python-crypto/blob/7069a33465307e2e100c3d91f15ac2b2fa084b85/solar_crypto/transactions/builder/base.py#L98
- https://github.com/Solar-network/python-crypto/blob/7069a33465307e2e100c3d91f15ac2b2fa084b85/solar_crypto/transactions/builder/base.py#L101
- https://github.com/Solar-network/python-crypto/blob/7069a33465307e2e100c3d91f15ac2b2fa084b85/solar_crypto/transactions/builder/base.py#L104
- https://github.com/Solar-network/python-crypto/blob/7069a33465307e2e100c3d91f15ac2b2fa084b85/solar_crypto/transactions/builder/base.py#L110
- https://github.com/Solar-network/python-crypto/blob/7069a33465307e2e100c3d91f15ac2b2fa084b85/solar_crypto/transactions/builder/base.py#L113
- https://github.com/Solar-network/python-crypto/blob/7069a33465307e2e100c3d91f15ac2b2fa084b85/solar_crypto/transactions/builder/base.py#L131
- https://github.com/Solar-network/python-crypto/blob/7069a33465307e2e100c3d91f15ac2b2fa084b85/solar_crypto/transactions/builder/base.py#L142

File: solar_crypto/transactions/builder/htlc_refund.py
Comment (sleepdefic1t) [inline solar_crypto/transactions/builder/htlc_refund.py:1]: missing return type:

https://github.com/Solar-network/python-crypto/blob/7069a33465307e2e100c3d91f15ac2b2fa084b85/solar_crypto/transactions/builder/htlc_refund.py#L27

File: solar_crypto/transactions/builder/ipfs.py
Comment (sleepdefic1t) [inline solar_crypto/transactions/builder/ipfs.py:1]: two missing return types:

- https://github.com/Solar-network/python-crypto/blob/7069a33465307e2e100c3d91f15ac2b2fa084b85/solar_crypto/transactions/builder/ipfs.py#L24
- https://github.com/Solar-network/python-crypto/blob/7069a33465307e2e100c3d91f15ac2b2fa084b85/solar_crypto/transactions/builder/ipfs.py#L27

File: solar_crypto/transactions/builder/transfer.py
Comment (sleepdefic1t) [inline solar_crypto/transactions/builder/transfer.py:1]: two missing return types:

- https://github.com/Solar-network/python-crypto/blob/7069a33465307e2e100c3d91f15ac2b2fa084b85/solar_crypto/transactions/builder/transfer.py#L28
- https://github.com/Solar-network/python-crypto/blob/7069a33465307e2e100c3d91f15ac2b2fa084b85/solar_crypto/transactions/builder/transfer.py#L31

File: solar_crypto/transactions/transaction.py
Comment (sleepdefic1t) [inline solar_crypto/transactions/transaction.py:1]: There are several functions throughout this file missing parameter and/or return types.

**Review by sleepdefic1t — CHANGES_REQUESTED** [review]
There are quite a few changes that will need to be made so that things are consistent, but overall this is very solid work.

It also appears that the 'black' formatter needs to be run, there are two files that change after linting.

Thank you and I look forward to the revisions.

File: solar_crypto/identity/private_key.py (hunk 1 above)
Comment (sleepdefic1t) [inline solar_crypto/identity/private_key.py:51]: The return type is great, but the parameter should probably be `str` too, shouldn't it?

File: solar_crypto/transactions/builder/base.py
Comment (sleepdefic1t) [inline solar_crypto/transactions/builder/base.py:1]: should `to_dict` and `to_json` also have `-> dict:` as a return type?

[the remark this answers is not in the document]
Comment (sleepdefic1t) [reply solar_crypto/transactions/builder/base.py:1]: Do the `verify` methods need return types too?

File: solar_crypto/identity/private_key.py (hunk 1 above)
Comment (c0nsol3) [reply solar_crypto/identity/private_key.py:51]: No, because it returns the `PrivateKey` object

**Review by sleepdefic1t — CHANGES_REQUESTED** [review]
I think there are only a couple/few very minor items left 👍 

Thank you again, @c0nsol3 !

**Review by sleepdefic1t — APPROVED** [review]

**Comment by sleepdefic1t** [comment]
Thank you, @c0nsol3 🎉

**Review by Cactus-15-49 — APPROVED** [review]

**Pull request merged**